### PR TITLE
refactor: improve code reuse

### PR DESCRIPTION
**Pull request by rubenporras**

### CODE REVIEW

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java [hunk 1]
```diff
@@ -341,6 +327,16 @@ private static IContextInformation toContextInformation(SignatureInformation inf
 		return new ContextInformation(information.getLabel(), signature.toString());
 	}
 
+	private void cancelFuture(@Nullable CompletableFuture<List<Void>> future) {
```
Comment (sebthom) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:330]: how about adding it as `public static void cancelFuture(@Nullable Future<?> future)` to `org.eclipse.lsp4e.internal.CancellationUtil` and reuse it in other places too?

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java [hunk 2]
```diff
@@ -216,40 +214,28 @@ private String createErrorMessage(int offset, Exception ex) {
...
 						}
 						return CompletableFuture.completedFuture(null);
 					});
-			this.contextInformationLanguageServersFuture = LanguageServers.forDocument(document)
+			contextInformationLanguageServersFuture = LanguageServers.forDocument(document)
 					.withFilter(capabilities -> capabilities.getSignatureHelpProvider() != null) //
 					.collectAll((w, ls) -> {
-						SignatureHelpOptions provider = castNonNull(w.getServerCapabilities()).getSignatureHelpProvider();
+						List<String> triggerChars = castNonNull(w.getServerCapabilities()).getSignatureHelpProvider().getTriggerCharacters();
 						synchronized (contextTriggerCharsSemaphore) {
```
Comment (sebthom) [inline org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:237]: I know, not part of your PR, but I think these synchronized blocks can be avoided too. mergeTriggers is a static method so no need for synchronization and contextTriggerChars can be assigned atomicly. I.e. declaring contextTriggerChars volatile should be enough and actually better.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java (hunk 1 above)
Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:330]: Following up on your comment, I cannot find any other place in the codebase that uses this idiom. I have them read the documentation of `CompletableFuture.cancel(boolean)` and checked the implementation, and I think the exception will never be thrown. So we can just remove the catch block. 

Do you agree?

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java (hunk 2 above)
Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:237]: Great, I will do that after I have an answer to the other thread in this review.

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java (hunk 1 above)
Comment (sebthom) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:330]: Yes. Reading the JavaDoc and checking the implementation it looks like it is never thrown. So removing the try catch should be ok

Comment (sebthom) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:324]: I still would suggest adding this as `public static void cancelFuture(@Nullable Future<?> future)` to `org.eclipse.lsp4e.internal.CancellationUtil` and reuse it in other places. There are a few more places where a future is first checked for null and then canceled and if we have a method here already why not use it everywhere else.

**Review by sebthom — APPROVED** [review]

File: org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java (hunk 1 above)
Comment (rubenporras) [reply org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/completion/LSContentAssistProcessor.java:324]: I will take a look.

**Review by sebthom — APPROVED** [review]

**Pull request merged**